src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,22 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.operation.OperationHandler;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public OperationHandler get(Operation operation) {
```

get(Operation operation) should validate the operation argument. Currently calling get(null) will cause unclear behavior; add `if (operation == null) throw new RuntimeException("Operation must not be null");` to fail fast and provide an informative message (matches checklist expectations to validate public inputs) .